feat(keybinds): improve keybind hook to allow using react state

JulienZD opened the pull request:
Previously you'd need to interact with state manually from jotai to have reactive keybinds

Unfortunately the `when` option now doesn't work anymore, since the state within would be stale. We can just check in the handler callbacks and return if we shouldn't act. Sadly this does mean that the keybind is always shown in the display, since that can't check for "enabled" keybinds anymore. Perhaps we can look into how to fix this in the future, but I don't think it's that big of a deal right now

JulienZD committed:
feat(keybinds): improve keybind hook to allow using react state

Previously you'd need to interact with state manually from jotai to
have reactive keybinds

JulienZD committed:
feat(keybinds): add string keybind parser

This should make it easier to add keybinds with modifiers

JulienZD commented on src/lib/keybinds/parse-keybind.test.ts:
Time for a CI that runs tests on PRs?

JulienZD committed:
feat(keybinds): rewrite all useKeybind usages to use new keybind parser

JulienZD committed:
chore: test in precommit

RobinHeidenis commented on src/board-view.tsx:
```diff
@@ -112,137 +113,172 @@ export const BoardView = () => {
+      name: "Filter users",
+    },
+    () => setSelectUsersModalOpen(true),
+    [],
```
is it possible to give this a default argument of`[]`? so if there's no deps we don't have to pass it?

RobinHeidenis replied:
oh it already does
```suggestion
```

JulienZD committed:
feat(keybinds): support running in floating nvim terminal

q is a commonly used keybind to close a floating terminal, so let's just unmap it.
Could always add it back using user configuration

JulienZD committed:
refactor(keybinds): remove empty deps array as it's already provided by default

JulienZD commented:
@RobinHeidenis Can I merge?
